benchmarks/src/main/scala/benchmark/MergeBenchmark.scala

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (2023) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package benchmark
+
+import java.util.UUID
+
+import org.apache.spark.sql.Row
+
+trait MergeConf extends BenchmarkConf {
+  def scaleInGB: Int
+  def tableName: String = "web_returns"
+  def userDefinedDbName: Option[String]
+  def dbName: String = userDefinedDbName.getOrElse(s"merge_sf${scaleInGB}")
+  def dbLocation: String = dbLocation(dbName)
+}
+
+case class MergeBenchmarkConf(
+     scaleInGB: Int = 0,
+     userDefinedDbName: Option[String] = None,
+     iterations: Int = 3,
+     benchmarkPath: Option[String] = None) extends MergeConf {
+}
+
+object MergeBenchmarkConf {
+  import scopt.OParser
+  private val builder = OParser.builder[MergeBenchmarkConf]
+  private val argParser = {
+    import builder._
+    OParser.sequence(
+      programName("Merge Benchmark"),
+      opt[String]("scale-in-gb")
+        .required()
+        .valueName("<scale of benchmark in GBs>")
+        .action((x, c) => c.copy(scaleInGB = x.toInt))
+        .text("Scale factor of the TPCDS benchmark"),
+      opt[String]("benchmark-path")
+        .required()
+        .valueName("<cloud storage path>")
+        .action((x, c) => c.copy(benchmarkPath = Some(x)))
+        .text("Cloud path to be used for creating table and generating reports"),
+      opt[String]("iterations")
+        .optional()
+        .valueName("<number of iterations>")
+        .action((x, c) => c.copy(iterations = x.toInt))
```

Comment:
Same about the conversion here.

Reply:
The current error is:
```
Error: Option --scale-in-gb expects a number but was given 'test'
```
which is pretty good already.